pkg/proctree/fork_processing.go

```diff
@@ -0,0 +1,176 @@
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+type taskIds struct {
+	Pid  int
+	Tid  int
+	Ppid int
+}
+
+// ProcessForkEvent add new process to process tree if new process created,
+// or update process threads if new thread created.
+// Because the fork at start is only a copy of the father,
+// the important information regarding of the process information and binary will be collected
+// upon execve.
+func (tree *ProcessTree) ProcessForkEvent(event *trace.Event) error {
+	err := tree.processGeneralEvent(event)
+	if err != nil {
+		return err
+	}
+
+	newHostIds, err := parseForkHostIds(event)
+	if err != nil {
+		return err
+	}
+	newNsIds, err := parseForNsIds(event)
+	if err != nil {
+		return err
+	}
+
+	if newHostIds.Pid == newHostIds.Tid {
+		err = tree.addForkProcess(event, newHostIds, newNsIds)
+		if err != nil {
+			return err
+		}
+	}
+	return tree.addForkThread(event, newHostIds, newNsIds)
+}
+
+// addForkProcess create a new process using fork event and the IDs given to the new process
+// Notice that the new process information is a duplicate of the father, until an exec will occur.
+func (tree *ProcessTree) addForkProcess(event *trace.Event, newInHostIds taskIds, newInNsIds taskIds) error {
+	newProcess, err := tree.getProcess(newInHostIds.Pid)
+	// If it is a new process or if for some reason the existing process is a result of lost exit
+	// event
+	if err != nil ||
+		newProcess.getForkTime() != 0 {
+		tree.immediateRemoveProcessFromTree(newInHostIds.Pid)
+		newProcess, err = tree.newForkedProcessNode(newInHostIds, newInNsIds)
+		if err != nil {
+			return err
+		}
+	}
+	newProcess.mutex.Lock()
+	tree.copyParentBinaryInfo(types.Timestamp(event.Timestamp), newProcess)
+
+	newProcess.setGeneralInfoProtected(
+		newInNsIds.Pid,
+		event.UserID,
+		event.Container.ID,
+	)
+
+	newProcess.setForkTime(types.Timestamp(event.Timestamp))
+	newProcess.mutex.Unlock()
+	return nil
+}
+
+// addForkThread create a new thread using fork event and the IDs given to the new thread
+func (tree *ProcessTree) addForkThread(event *trace.Event, newInHostIds taskIds, newInNsIds taskIds) error {
+	process, err := tree.getProcess(newInHostIds.Pid)
+	if err != nil {
+		return err
+	}
+	process.mutex.Lock()
+	newThread, err := tree.newProcessThreadNode(process, newInHostIds.Tid)
+	if err != nil {
+		process.mutex.Unlock()
+		return err
+	}
+
+	processExitTime := process.getExitTime()
+	process.mutex.Unlock()
+	newThread.mutex.Lock()
+	newThread.setForkTime(types.Timestamp(event.Timestamp))
+	newThread.setName(types.Timestamp(event.Timestamp), event.ProcessName)
+	newThread.setGeneralInfoProtected(
+		newInNsIds.Tid,
+		event.ProcessName,
+		namespacesIds{
+			pid:   event.PIDNS,
+			mount: event.MountNS,
+		},
+		processExitTime,
+	)
+	newThread.mutex.Unlock()
+	return nil
+}
+
+// parseForkHostIds gets the new forked process Ids in the host PId namespace
+func parseForkHostIds(event *trace.Event) (taskIds, error) {
```

Comment:
I believe everything that is "PID" should be named PID and not ID. It makes things confusing reading ID when we are able to know if its the PID/TID right away by reading the function/variable name.